Interrogator.xUnit/NotConcurrentAttribute.cs

```diff
@@ -0,0 +1,95 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using Interrogator.xUnit.Common;
+using Interrogator.xUnit.Utilities;
+
+namespace Interrogator.xUnit
+{
+	[AttributeUsage(AttributeTargets.Method, AllowMultiple = true)]
+	public class NotConcurrentAttribute : DependsOnAttribute
+	{
+		public enum ConcurrencyScope
+		{
+			Class,
+			Namespace,
+			Assembly
+		}
+
+		public string GroupName { get; }
+		public ConcurrencyScope Scope { get; }
+		public string CurrentMethodName { get; }
+
+		public NotConcurrentAttribute(string groupName, [CallerMemberName]string currentMethodName = null) : this(groupName, ConcurrencyScope.Class, currentMethodName) { }
+
+		public NotConcurrentAttribute(string groupName, ConcurrencyScope scope, [CallerMemberName]string currentMethodName = null) : base(null)
+		{
+			GroupName = $"{groupName}_{scope}"; // Adding scope to group name ensures that different scoped attributes don't collide
+			Scope = scope;
+			CurrentMethodName = currentMethodName;
+		}
+
+		internal override Result<Option<MethodInfo>, string> TryGetMethod(Type containingType)
```

Comment:
I think I'd rather this get modified to have the Member this was attached to passed in. Then you can get rid of CurrentMethodName